### ISSUE TITLE
KotlinGenerator generates invalid code when column names X and isX clash

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11912]
The `KotlinGenerator` produces problematic code for tables like:

```sql
CREATE TABLE t (
  x INT,
  is_x INT
);
```

E.g.

```kotlin
data class T(
  var x: Int?,
  var isX: Int?
)
```

We should be generating this, instead:

```kotlin
data class T(
  var x: Int?,
  @set:JvmName("setIsX") var isX: Int?
)
```

----
See also:
https://stackoverflow.com/q/67698228/521799

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.12 (#11913)

**Comment by lukaseder** [comment]
The backported version will remain as it is, but in jOOQ 3.15, I wonder if we should simply generate this annotation for all columns starting with `IS_` in kotlin? I don't think the kotlin status quo is *ever* desirable, especially from an interop perspective.

I'll change this fix as follows:

- [ ] Add a configuration for the annotation, defaulting to `true`
- [ ] Apply the annotation on *all* setters for columns starting with `IS_`

See also: https://stackoverflow.com/q/67898472/521799

**Comment by lukaseder** [comment]
Improved fix integrated for jOOQ 3.15

**Comment by lukaseder** [comment]
Related: https://github.com/jOOQ/jOOQ/issues/15850
